crates/sui-types/src/gas.rs

```diff
@@ -576,6 +577,9 @@ pub fn check_gas_balance(
     for extra_obj in more_gas_objs {
         gas_balance += get_gas_balance(&extra_obj)? as u128;
     }
+    if !protocol_config.gas_checks_v2() {
+        required_gas_amount = (gas_budget as u128) * (gas_price as u128);
+    }
```

Comment:
if i'm not mistaken, this function (check_gas_balance) is also called in the execution path, right? so this would have been an execution bug as well?

Reply:
correct, as I thought we concluded on the slack channel.
The idea that signing would have saved us was completely bogus.